### PR TITLE
Kernel: Switch more AK::String usages to KString

### PR DESCRIPTION
**Pull request by bgianfo**

### CODE REVIEW

**Comment by awesomekling** [comment]
>`AK::KString`

`KString` is not in the `AK` namespace :)

**Comment by supercomputer7** [comment]
The last changes today of the PCI subsystem simplification made this to have lots of conflicts, sorry about that.

Nevertheless, very cool changes. I might have some questions, but feel free to resolve them everything seems clear to you :)

File: Kernel/Net/NetworkAdapter.h [hunk 1]
```diff
@@ -50,7 +50,7 @@ class NetworkAdapter : public RefCounted<NetworkAdapter>
 
     virtual StringView class_name() const = 0;
 
-    const String& name() const { return m_name; }
+    const StringView name() const { return m_name->view(); }
```
Comment (supercomputer7) [inline Kernel/Net/NetworkAdapter.h:53]: Isn't that supposed to be a `const StringView&` return value? what is the meaning of `const StringView`?

Comment (bgianfo) [reply Kernel/Net/NetworkAdapter.h:53]: it's supposed to just be `StringView`, KString::view returns a new StringView, so this can't be a reference.

**Pull request merged**